### ISSUE TITLE
0.0.14 - AttributeError: 'NoneType' object has no attribute 'state'

### ISSUE THREAD

**Issue by Boss-man** [humbertogontijo/homeassistant-roborock#84]
After a while I get this error and it repeat like every 30 seconds.

```
2023-01-02 14:00:17.363 DEBUG (MainThread) [custom_components.roborock.api.api] id=3286 Requesting method get_status with None
2023-01-02 14:00:17.364 DEBUG (MainThread) [custom_components.roborock.api.api] id=3287 Requesting method get_dnd_timer with None
2023-01-02 14:00:17.366 DEBUG (MainThread) [custom_components.roborock.api.api] id=3288 Requesting method get_clean_summary with None
2023-01-02 14:00:17.367 DEBUG (MainThread) [custom_components.roborock.api.api] id=3289 Requesting method get_consumable with None
2023-01-02 14:00:21.365 WARNING (MainThread) [custom_components.roborock.api.api] Timeout after 4 seconds waiting for get_status response
2023-01-02 14:00:21.367 WARNING (MainThread) [custom_components.roborock.api.api] Timeout after 4 seconds waiting for get_dnd_timer response
2023-01-02 14:00:21.368 WARNING (MainThread) [custom_components.roborock.api.api] Timeout after 4 seconds waiting for get_clean_summary response
2023-01-02 14:00:21.369 WARNING (MainThread) [custom_components.roborock.api.api] Timeout after 4 seconds waiting for get_consumable response
2023-01-02 14:00:21.371 DEBUG (MainThread) [custom_components.roborock] Finished fetching roborock data in 4.008 seconds (success: True)
2023-01-02 14:00:21.371 ERROR (MainThread) [homeassistant] Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 168, in _handle_refresh_interval
    await self._async_refresh(log_failures=True, scheduled=True)
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 316, in _async_refresh
    self.async_update_listeners()
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 121, in async_update_listeners
    update_callback()
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 381, in _handle_coordinator_update
    self.async_write_ha_state()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 545, in async_write_ha_state
    self._async_write_ha_state()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 583, in _async_write_ha_state
    state = self._stringify_state(available)
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 551, in _stringify_state
    if (state := self.state) is None:
  File "/config/custom_components/roborock/vacuum.py", line 296, in state
    state = self._device_status.state
AttributeError: 'NoneType' object has no attribute 'state'
```

**Comment by shred86** [comment]
This is related to #41 

When we're timing out from receiving anything from Roborock's MQTT broker, it's setting all the device information to a `None` type. Once the timeouts are handled, then we'll just need to make sure we're not overriding the existing HA device information in these timeout situations.

**Comment by Boss-man** [comment]
Seems to be fixed between 0.0.15 - 0.0.17
